sale_timesheet_force_non_billable/models/account_analytic_line.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2019-Today: La Louve (<https://cooplalouve.fr>)
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    force_non_billable = fields.Boolean(
+        default=False,
+        track_visibility='onchange'
+    )
+
+    @api.multi
+    @api.depends('so_line.product_id', 'project_id', 'task_id',
+                 'force_non_billable')
+    def _compute_timesheet_invoice_type(self):
+        timesheets = self.filtered(lambda t: t.force_non_billable)
+        timesheets.update({'timesheet_invoice_type': 'non_billable'})
+        return super(AccountAnalyticLine, self - timesheets).\
+            _compute_timesheet_invoice_type()
+
+    @api.multi
+    def set_non_billable(self):
+        for line in self:
+            line.force_non_billable = True
+        return True
+
+    @api.multi
+    def remove_non_billable(self):
+        for line in self:
+            line.force_non_billable = False
+        return True
+
+    @api.multi
+    def toggle_non_billable(self):
+        for line in self:
+            line.force_non_billable = not line.force_non_billable
+        return True
```

Comment:
You can avoid having to implement these actions with the boolean_toggle widget
```
widget="boolean_toggle"
```

It looks like this on tree views, without having to implement buttons ;)
![image](https://user-images.githubusercontent.com/1914185/65688523-98ba9d00-e06b-11e9-81c9-a612f25d1df7.png)

Reply:
Ohh... that's what I was looking for, but maybe I added bool field in button tag, so it wasn't visible to me